packages/next/src/server/lib/start-server.ts

```diff
@@ -38,6 +39,65 @@ import type { ConfiguredExperimentalFeature } from '../config'
 const debug = setupDebug('next:start-server')
 let startServerSpan: Span | undefined
 
+/**
+ * Get the process ID (PID) of the process using the specified port
+ */
+async function getProcessIdUsingPort(port: number): Promise<string | null> {
+  const timeoutMs = parseInt(
+    process.env.NEXT_PID_DETECTION_TIMEOUT || '250',
+    10
+  )
```

Comment:
I would prefer to not make it configureable. That hints at 250 not being the right value. How long did it take on your systems?